flutter/lib/src/screenshot/recorder.dart

```diff
@@ -110,34 +191,28 @@ class ScreenshotRecorder {
         image.dispose();
       }
 
-      if (filter != null) {
-        _obscureWidgets(canvas, filter.items);
+      if (obscureItems != null) {
+        _obscureWidgets(canvas, obscureItems);
       }
 
       final picture = recorder.endRecording();
+      Timeline.finishSync(); // Sentry::renderScreenshot
 
       try {
-        final finalImage = await picture.toImage(
-            (srcWidth * pixelRatio).round(), (srcHeight * pixelRatio).round());
-        _log(
-            SentryLevel.debug,
-            "captured a screenshot in ${watch.elapsedMilliseconds}"
-            " ms ($blockingTime ms blocking).");
+        Timeline.startSync('Sentry::screenshotToImage', flow: flow);
+        final finalImage = await picture.toImage(width, height);
+        Timeline.finishSync(); // Sentry::screenshotToImage
         try {
-          await callback(finalImage);
+          Timeline.startSync('Sentry::screenshotCallback', flow: flow);
+          _completer.complete(await callback(finalImage));
+          Timeline.finishSync(); // Sentry::screenshotCallback
         } finally {
-          finalImage.dispose(); // image needs to be disposed manually
+          finalImage.dispose(); // image needs to be disposed-of manually
```

Comment:
do we need a  `_completer.complete(null)` in finally if `await callback(finalImage)` throws?